tests/profiling/collector/test_stack_asyncio.py

```diff
@@ -37,6 +39,11 @@ async def hello() -> None:
         maintask = loop.create_task(hello(), name="main")
     else:
         maintask = loop.create_task(hello())
+
+    # Make sure we spend enough time in the main task (event.name=None) in MainThread for profile events to be collected
+    # So we wait for twice the collection time to account for small variances in time.sleep()
+    time.sleep(2 * _default_min_interval_time())
```

Comment:
It makes sense to me that increasing the collection time will fix flakiness caused by inadequate collection, but I'd like to discuss the explanation a bit.  If I say something agreeable, would you update the comment accordingly?

AFAIK, `time.sleep()` as of Python 3.5 will auto-restart if the runtime catches a signal.  It also makes the standard provision that the amount slept will be _at least_ as long as the amount specified, possibly longer.  What this means is, the `time.sleep()` on line 45 will be at _least_ as long as two minimum interval times.  In other words, time-variation of this sleep favors "more collection" than would be a priori expected from just the specified times/frequencies alone.

On the other hand, I think what's actually happening here is that rather than getting short-sleeps from this `time.sleep()`, we're actually getting _long_ collection intervals from `ddtrace/profiling/collector/stack.pyx:StackCollector.collect()`.  `collect()` will exacerbate any case where the scheduling mechanism _also_ provides that the sleep is "at least as long" as the one specified, since it'll only collect if the total waited time exceeds the specified interval (and you'll practically never sleep right _on_ the interval).

If this is accurate, then I think we should specify that we're accounting for variations in the collection sampling interval.

Reply:
I will update the comment, you're hi-lighting the correct behavior!